Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -343,44 +351,127 @@ namespace Terrain
+                        selectedSectors.at(selectedIndex) = true;
+
+                        if (stackIndex == 0)
+                        {
+                            // Since this is the first stack, no previous stack to check, so just draw.
+                            m_sectorsToDraw.push_back({ sector.m_aabb, sector.m_rhiDrawPacketClod.get() });
+                            continue;
+                        }
+
+                        Vector2i previousCoord = (sector.m_worldCoord - sectorStack.m_startCoord) * 2;
+                        uint32_t previousDrawnIndex = previousCoord.m_y * nextLodSectorCount + previousCoord.m_x;
+
+                        // Check the 4 sectors in the previous LOD that are covered by this sector.
+                        uint8_t coveredByHigherLod =
+                            (uint8_t(previousSelectedSectors.at(previousDrawnIndex))                          << 0) | // Top left
+                            (uint8_t(previousSelectedSectors.at(previousDrawnIndex + 1))                      << 1) | // Top right
+                            (uint8_t(previousSelectedSectors.at(previousDrawnIndex + nextLodSectorCount))     << 2) | // Bottom left
+                            (uint8_t(previousSelectedSectors.at(previousDrawnIndex + nextLodSectorCount + 1)) << 3);  // Bottom right
+
+                        if (coveredByHigherLod == 0b1111)
+                        {
+                            continue; // Completely covered by previous LOD, so do nothing
+                        }
+                        if (coveredByHigherLod == 0b0000)
+                        {
+                            // Not covered at all by previous LOD, so the draw entire sector
+                            m_sectorsToDraw.push_back({ sector.m_aabb, sector.m_rhiDrawPacketClod.get() });
+                        }
+                        else
+                        {
+                            // Partially covered by previous LOD. Draw only missing quadrants
+                            for (uint8_t i = 0; i < 4; ++i)
+                            {
+                                if ((coveredByHigherLod & 0b0001) == 0b0000)
+                                {
+                                    m_sectorsToDraw.push_back({ sector.m_quadrantAabbs.at(i), sector.m_rhiDrawPacketQuadrant.at(i).get() });
+                                }
+                                coveredByHigherLod >>= 1;
+                            }
+                        }
                     }
                 }
-                minDistanceSq = maxDistanceSq;
+
                 maxDistanceSq = maxDistanceSq * 4.0f; // Double the distance with squared distances is * 2^2.
+                previousSelectedSectors = AZStd::move(selectedSectors);
             }
         }
 
-        m_previousCameraPosition = mainCameraPosition;
+        // Second pass - compare views against the list of selected sectors.
```

Comment:
It's not just _comparing_ the views, it's also _adding_ the draw packets if they're visible.

Reply:
will update the comment.